src/pages/my/todo.tsx

```diff
@@ -1,30 +1,119 @@
 import MyHomeLayout from '../../components/layout/MyHomeLayout';
 import { PageWithLayout } from '../../types';
 import { scaffold } from '../../utils/next';
+import { useRespondEvents } from '../../hooks';
 
 const scaffoldOptions = {
     localeScope: [
         'layout.my',
+        'misc.callAssignmentList',
+        'misc.eventList',
         'misc.publicHeader',
         'pages.myTodo',
     ],
 };
 
-export const getServerSideProps : GetServerSideProps = scaffold(async () => {
-    return {
-        props: {},
-    };
+export const getServerSideProps : GetServerSideProps = scaffold(async (context) => {
+    const { user } = context;
+
+    let respondEventsState;
+    let bookedEventsState;
+    let callAssignmentsState;
+
+    if (user) {
+        await context.queryClient.prefetchQuery('respondEvents', getRespondEvents(context.apiFetch));
+        await context.queryClient.prefetchQuery('bookedEvents', getBookedEvents(context.apiFetch));
+        await context.queryClient.prefetchQuery('callAssignments', getCallAssignments(context.apiFetch));
+
+        respondEventsState = context.queryClient.getQueryState('respondEvents');
+        bookedEventsState = context.queryClient.getQueryState('bookedEvents');
+        callAssignmentsState = context.queryClient.getQueryState('callAssignments');
+    }
+
+    if (respondEventsState?.status === 'success'
+        && bookedEventsState?.status === 'success'
+        && callAssignmentsState?.status === 'success') {
+
+        return {
+            props: {},
+        };
+    }
+    else {
+        return {
+            notFound: true,
+        };
+    }
 }, scaffoldOptions);
 
 const MyTodoPage : PageWithLayout = () => {
+    const bookedEventsQuery = useQuery('bookedEvents', getBookedEvents());
+    const callAssignmentsQuery = useQuery('callAssignments', getCallAssignments());
+
+    const { respondEvents, onUndoSignup } = useRespondEvents();
+
+    if ((!respondEvents || respondEvents.length === 0)
+        && (!callAssignmentsQuery.data || callAssignmentsQuery.data?.length === 0)) {
+        return (
+            <Text data-testid="no-events-placeholder">
+                <Msg id="pages.myTodo.placeholder"/>
+            </Text>
+        );
+    }
+
+    if (!callAssignmentsQuery.data || callAssignmentsQuery.data?.length === 0) {
+        return (
+            <>
+                <Heading level={ 1 }>
+                    <Msg id="pages.myTodo.heading"/>
+                </Heading>
+                <View marginBottom="size-500">
+                    <Heading level={ 2 } marginBottom="0">
+                        <Msg id="pages.myTodo.events"/>
+                    </Heading>
+                    <EventList
+                        bookedEvents={ bookedEventsQuery.data }
+                        events={ respondEvents }
+                        onUndoSignup={ onUndoSignup }
+                    />
+                </View>
+            </>
```

Comment:
There is a lot of code duplication between this and lines 94-116 below (the default JSX return value). Can you refactor this to make it more concise?